src/run.rs

```diff
@@ -539,6 +546,25 @@ fn vacuum(state: &mut State, threshold: &Byte) -> io::Result<()> {
             .then(y.ancestors.cmp(&x.ancestors))
     });
 
+    // If the user provided the keep argument, we need to filter out those images which match the
+    // provided regexes.
+    if let Some(regex_set) = keep {
+        sorted_image_nodes = sorted_image_nodes
+            .into_iter()
+            .filter(|image_node| {
+                if regex_set.is_match(&image_node.image_info.repository) {
+                    info!(
+                        "Skipping image {} which matches one or more of the provided regexs",
+                        image_node.image_info.repository
+                    );
+                    return false;
```

Comment:
If you don't mind, I have a slight preference for formatting the lambda like this (instead of the imperative-style `return`):

```rust
if regex_set.is_match(&image_node.image_info.repository) { {
    info!(...);

    false
} else {
    true
}
```